src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```diff
@@ -314,10 +315,23 @@ public ConnectionMetadata getConnectionMetadata() {
     return connectionMetadata;
   }
 
+  public synchronized IntermediateStatement getActiveStatement() {
+    return activeStatementsMap.get(this.connectionId);
+  }
+
+  public synchronized ConnectionStatus getStatus() {
+    return status;
+  }
+
+  public synchronized void setStatus(ConnectionStatus status) {
+    this.status = status;
+  }
+
   /** Status of a {@link ConnectionHandler} */
-  private enum ConnectionStatus {
+  public enum ConnectionStatus {
```

Comment:
why does this need to be public?

Reply:
Needed for a later change.  I'll revert for now.